### ISSUE TITLE
Support for flashing opensource firmwares (esphome, tasmota, etc.)

### ISSUE THREAD

**Issue by imduffy15** [tuya/tuya-home-assistant#95]
In the past we used to be able to flash opensource firmware to our ESP8266 devices, this was amazing as it allowed us to completely remove any internet dependency.

Sadly, due to changes, the method for doing this defined at https://github.com/ct-Open-Source/tuya-convert no longer works.

Could we please get an official (warranty is definitely void if you do it) method for converting Tuya based ESP8266 devices to tasmota, esphome and others? not having an official method causes us to have to open up devices and attach to the different pins on the PCB, in some instances this means breaking the device, thus it not making sense to buy a Tuya device in the first place.

In the past I previously bought from https://www.novostella.net/ and flashed accordingly, however I have stopped doing this as I would have been stuck with the Tuya firmware. I now buy https://www.athom.tech/ which comes preflashed but the selection is more limited. 

I'm  not alone on  this one, there is  a full list of devices at https://github.com/ct-Open-Source/tuya-convert/wiki/Collaboration-document-for-PSK-Identity-02#known-affected-devices that people would love to flash with opensource firmwares.

**Comment by zlinoliver** [comment]
@imduffy15  Sorry, flashing PBT smart device's firmware conflicts with the Tuya business model, we won't provide any help or support to these methods.

The Tuya Home Assistant will support local control in Q3, 2021, you can achieve local control feature easily with the update.

**Comment by imduffy15** [comment]
How does enable changing the firmware impact on the business model? With local control enabled I would hope to put any Tuya devices on a network with no internet. They should be unknown to Tuya given no internet, if they are unknown what is the harm in allowing changing firmwares? A changed firmware device would be similarly unknown. 

I would understand that the action of changing firmware would void any warranty or support but I believe most people would accept that.

**Comment by controlzone** [comment]
I agree, especially since we would still be buying the hardware side of things. 
Even if there was an initial activation to register the product being sold / used and then having it's intended purpose to be re-flashed, i would have thought Tuya would like to know this info so that they could improve the functionality so that perhaps reflashing wasnt necessary. 
Not allowing firmware modification or alternative firmware just makes me want to find a completely different product and ecosystem - this would hurt Tuya more surely.....?

**Comment by piwi3910** [comment]
As much i would like to see it being easier, and not blocked on newer firmwares it does makes sense.
Tuya doesn't make HW. The product vendors do. And they pay Tuya for use of their Platform.
So if we flash other firmwares, we don't use Tuya's platform, and so they don't get paid by the HW vendors
